### PR TITLE
chore: Add MonoToFuture to reduce conversion.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
`toFuture.asScala` seems quite boring, we can convert it to Scala Future directly.

Result:
Less allocation/conversion

### CODE REVIEW

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFuture.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
```
Comment (pjfanning) [inline core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFuture.scala:5]: is it possible that we should be using the standard ASF header here? this Akka derived header is only for Akka derived code

File: core/src/test/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFutureSpec.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
```
Comment (pjfanning) [inline core/src/test/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFutureSpec.scala:5]: I think this should have the standard ASF header

File: core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFuture.scala (hunk 1 above)
Comment (He-Pin) [reply core/src/main/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFuture.scala:5]: Will update later

File: core/src/test/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFutureSpec.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import org.scalatest.TestSuite
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import reactor.core.publisher.Mono
+
+class MonoToFutureSpec extends AnyWordSpec with ScalaFutures with TestSuite with Matchers {
+  "MonoToFutureSpec" should {
+    "convert a Mono to a Future" in {
+      val r = Mono.just("pekko")
```
Comment (laglangyue) [inline core/src/test/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFutureSpec.scala:21]: hard string.
Adding test for empty and seq stream should be better . I am not very similar with Mono.

**Comment by laglangyue** [comment]
BTW, I have a question: 

if a server provide a java-client using java completeFuture and a mono/flux client. Which should I use in pekko-streams api?

use completeFuture client, I can writer code `Source.future(completeFuture.asScala)` , or use the other like this PR

File: core/src/test/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFutureSpec.scala (hunk 3 above)
Comment (He-Pin) [reply core/src/test/scala/org/apache/pekko/persistence/r2dbc/internal/MonoToFutureSpec.scala:21]: Mono is just some kind of Future, which can only emit one/zero elements.

**Comment by He-Pin** [comment]
@laglangyue You can convert a Publisher to Pekko's Source, and a Pekko's Sink as a Publisher too, I mostly using CompletableFuture at $Work, but will use Pekko Stream when it fit best.

**Comment by pjfanning** [comment]
Generally looks ok but I'd prefer not to merge this till we split off a 1.0.x branch.

**Comment by laglangyue** [comment]
LGTM

**Review by laglangyue — APPROVED** [review]

**Review by jxnu-liguobin — APPROVED** [review]

**Review by mdedetrich — APPROVED** [review]
lgtm

I think that the inliner (which I have to enable now) would also solve the same problem but its only available for scala 2

**Review by pjfanning — APPROVED** [review]
lgtm - as @mdedetrich we actually can merge this because we have branched 1.0.x already

**Comment by He-Pin** [comment]
Thanks.
